src/app/main/time-groups/edit-time-group/edit-time-group.component.ts

```diff
@@ -212,28 +229,110 @@ export class EditTimeGroupComponent implements OnInit {
     this.subscriptions.push(
       this.timeGroupService.updateTimeGroupEmployees(this.timeGroupId, updateParams).subscribe(response => {
         this.appNotificationService.push(this.translate.instant('tr_time_group_employees_updated_successfully'), 'success');
+        this.employeesList = []
+        this.getEmployees()
       }, error => {
         this.appNotificationService.push(error.error.name, 'error');
         this.isFormSubmitted = false;
       }))
   }
+  checkIsInsideCurrentTimeGroup(employee: Employee) {
+    return employee.isInsideCurrentTimeGroup
+  }
   public assignEmployeeIngroup(event: any, timegroupEmployee: Employee) {
+
     if (event.target.checked) {
-      this.timeGroup.employees?.push(timegroupEmployee)
+      if (timegroupEmployee.time_group && timegroupEmployee.time_group.id != this.timeGroup.id) {
+        let data = {
+          title: this.translate.instant("tr_unassign_message"),
+          buttons: [
+            {
+              label: this.translate.instant("tr_action.cancel"),
+              actionCallback: 'cancel',
+              type: 'btn-secondary'
+            },
+            {
+              label: this.translate.instant("tr_action.ok"),
+              actionCallback: 'ok',
+              type: 'btn-primary'
+            }
+          ]
+        }
+        const dialogRef = this.dialog.open(SkoleraConfirmationComponent, {
+          width: '400px',
+          data: data,
+          disableClose: true
+        });
+        dialogRef.afterClosed().subscribe(result => {
+          if (result == 'ok') {
+            let params = {
+              "employee": {
+                "time_group_id": null
+              }
+            }
+            this.subscriptions.push(this.employeesService.updateEmployee(timegroupEmployee.id, params).subscribe((response: any) => {
+              this.appNotificationService.push(this.translate.instant('tr_unassign_time_group_successfully'), 'success');
+              timegroupEmployee.isInsideCurrentTimeGroup = true
+              this.timeGroup.employees?.push(timegroupEmployee)
+              this.filterTimeGroupEmployees()
+              this.employeesList = [];
+              this.getEmployees();
```

Comment:
Why you are not adding type = 'search' instead of `this.employeesList = []`

Reply:
done